xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -2165,6 +2160,17 @@ std::string CGUIMediaWindow::GetStartFolder(const std::string &dir)
       StringUtils::EqualsNoCase(dir, "root"))
     return "";
 
+  // Let plugins handle their own urls themselves
+  if (StringUtils::StartsWith(dir, "plugin://"))
+    return dir;
+
+  // Hack for Android items (numbered id's) on the leanback screen
```

Comment:
I don't know if it's correct to have Android specific code here but if it is required it should be at least `#if defined(TARGET_ANDROID)` to make it clear it is only for one platform.

Reply:
Originally this was a totally undocumented hack by @koying https://github.com/xbmc/xbmc/commit/01dbdf956f47e1df3c96261d154848b673fe996a#diff-05be91542194f697ea542d6c9db7d391R477

After investigating it turned out this was intended for the Android leanback screen. I'm not familiar with this functionality for Android. So, I kept the changes as small as possible to avoid breaking stuff.